api/hasura/actions/_handlers/endEpoch.ts

```diff
@@ -0,0 +1,140 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import { DateTime, Settings } from 'luxon';
+import { z } from 'zod';
+
+import { authCircleAdminMiddleware } from '../../../../api-lib/circleAdmin';
+import { adminClient } from '../../../../api-lib/gql/adminClient';
+import { errorResponseWithStatusCode } from '../../../../api-lib/HttpError';
+import { composeHasuraActionRequestBody } from '../../../../src/lib/zod';
+import { endEpochHandler } from '../../cron/epochs';
+
+Settings.defaultZone = 'utc';
+
+const EpochEndSchema = z.object({
+  id: z.number(),
+  circle_id: z.number(),
+});
+
+async function handler(request: VercelRequest, response: VercelResponse) {
+  const {
+    input: { payload: input },
+  } = composeHasuraActionRequestBody(EpochEndSchema).parse(request.body);
+
+  const deletingEpoch = await getExistingEpoch(input);
```

Comment:
@sshmm this is just being ended, not deleted, right?
```suggestion
  const endingEpoch = await getExistingEpoch(input);
```